homeassistant/components/swidget/entity.py

```diff
@@ -0,0 +1,59 @@
+"""Common code for Swidget."""
+
+from __future__ import annotations
+
+from collections.abc import Awaitable, Callable, Coroutine
+from typing import Any, Concatenate, TypeVar
+
+from swidget.swidgetdevice import SwidgetDevice
+from typing_extensions import ParamSpec
+
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import SwidgetDataUpdateCoordinator
+
+_T = TypeVar("_T", bound="CoordinatedSwidgetEntity")
+_P = ParamSpec("_P")
+
+
+def async_refresh_after(
+    func: Callable[Concatenate[_T, _P], Awaitable[None]],
+) -> Callable[Concatenate[_T, _P], Coroutine[Any, Any, None]]:
+    """Define a wrapper to refresh after."""
+
+    async def _async_wrap(self: _T, *args: _P.args, **kwargs: _P.kwargs) -> None:
+        await func(self, *args, **kwargs)
+        # await asyncio.sleep(1)
```

Comment:
appears unused, also the generics (typevars) appear to be unused, so please remove them until you actually use them